client/my-sites/upgrades/components/form/state-select.jsx

```diff
@@ -36,47 +36,56 @@ module.exports = React.createClass( {
 		} else {
 			this.refs.state.focus();
 		}
-	},
-
-	render: function() {
-		const classes = classNames( this.props.additionalClasses, 'state' ),
-			statesList = this.props.statesList.getByCountry( this.props.countryCode );
-		let options = [];
-
-		if ( isEmpty( statesList ) ) {
-			return (
-				<Input ref="state" { ...this.props } />
-			);
-		}
-
-		options.push( { key: '', label: this.translate( 'Select State' ), disabled: 'disabled' } );
+	}
 
-		options = options.concat( statesList.map( function( state ) {
-			if ( ! state.code ) {
-				return { key: '--', label: '', disabled: 'disabled' };
-			}
-			return { key: state.code, label: state.name };
-		} ) );
+	render() {
+		const classes = classNames( this.props.additionalClasses, 'state' );
 
 		return (
-			<div className={ classes }>
-				<div>
-					<FormLabel htmlFor={ this.props.name }>{ this.props.label }</FormLabel>
-					<FormSelect
-						ref="input"
-						name={ this.props.name }
-						value={ this.props.value }
-						disabled={ this.props.disabled }
-						onChange={ this.props.onChange }
-						onClick={ this.recordStateSelectClick }
-						isError={ this.props.isError } >
-						{ options.map( function( option ) {
-							return <option key={ option.key } value={ option.key } disabled={ option.disabled }>{ option.label }</option>;
-						} ) }
-					</FormSelect>
-				</div>
+			<div>
+				{ this.props.countryCode && <QueryCountryStates countryCode={ this.props.countryCode } /> }
+				{ isEmpty( this.props.countryStates )
+					? <Input ref="input" { ...this.props } />
+					: <div className={ classes }>
+						<FormLabel htmlFor={ this.props.name }>{ this.props.label }</FormLabel>
```

Comment:
Well, in that case the `name` would need to be a required prop, otherwise the `htmlFor` will be ineffective if omitted. Seems like a nuisance from an implementer's perspective. I personally think we should go the nesting route even if it requires style updates.